Proper error on bad attribute access in ms list

devoid opened the pull request:
- Fixes issue #50: Error message when using with option
- Now passing an invalid attribute to ms list returns
  a CLI error including a list of valid attributes; e.g.

```
$ ms list biochemistry/sdevoid/web/compounds -w idd
Invalid attribute 'idd' for ModelSEED::MS::Compound, available attributes:
        isCofactor
        defaultCharge
        ancestor_uuid
        abstractCompound
        deltaG
        uuid
        mapped_uuid
        pks
        formula
        mass
        modDate
        id
        unchargedFormula
        abstractCompound_uuid
        searchnames
        deltaGErr
        compoundCues
        parent
        name
        cksum
        structures
        comprisedOfCompounds
        comprisedOfCompound_uuids
        abbreviation
```
- Implement new exception 'ModelSEED::Exception::InvalidAttribute'
  where the throw arguments are:

``` perl
ModelSEED::Exception::InvalidAttribute->throw(
    object => $object,
    invalid_attribute => $attr,
);
```

Where `$object` is a `ModelSEED::MS` object and `$attr` is a string.

devoid committed:
Proper error on bad attribute access in ms list

- Fixes issue #50: Error message when using with option
- Now passing an invalid attribute to ms list returns
  a CLI error including a list of valid attributes; e.g.

    $ ms list biochemistry/sdevoid/web/compounds -w idd
    Invalid attribute 'idd' for ModelSEED::MS::Compound, available attributes:
            isCofactor
            defaultCharge
            ancestor_uuid
            abstractCompound
            deltaG
            uuid
            mapped_uuid
            pks
            formula
            mass
            modDate
            id
            unchargedFormula
            abstractCompound_uuid
            searchnames
            deltaGErr
            compoundCues
            parent
            name
            cksum
            structures
            comprisedOfCompounds
            comprisedOfCompound_uuids
            abbreviation

- Implement new exception 'ModelSEED::Exception::InvalidAttribute'
  where the throw arguments are:

    ModelSEED::Exception::InvalidAttribute->throw(
        object => $object,
        invalid_attribute => $attr,
    );

Where `$object` is a `ModelSEED::MS` object and `$attr` is a string.

pfrybar commented:
So this works, but when I do 'ms list biochemistry'' I now get this error:

Can't locate object method "ref" via package "biochemistry/pfrybarger/main" (perhaps you forgot to load "biochemistry/pfrybarger/main"?) at /home/paul/work/ModelSEED/lib/ModelSEED/App/mseed/Command/list.pm line 153.

devoid committed:
Fixed that, good catch.

- Fixed error with $ ms list biochemistry
- Still a bug with $ ms list biochemistry/sdevoid
  which comes from ModelSEED::Reference, basically there's not a
  good syntax for getting all of sdevoid's biochemistry objects
  given that kind of reference--I think it's even getting interpreted
  as a UUID, which is shocking.
